File: codes/julia/007_functions.jl
# ? function statement or function declaration

function get_rectangle_area_v1(rectangle_width::Any, rectangle_length::Any)::Any
    return (rectangle_width * rectangle_length)::Any
end
println("get_rectangle_area_v1(7, 5): $(get_rectangle_area_v1(7, 5))")

get_rectangle_area_v2(rectangle_width::Any, rectangle_length::Any)::Any = (rectangle_width * rectangle_length)::Any
println("get_rectangle_area_v2(7, 5): $(get_rectangle_area_v2(7, 5))")

# ? function expression

get_rectangle_area_v3::Any = function (rectangle_width::Any, rectangle_length::Any)
    return (rectangle_width * rectangle_length)::Any
end
println("get_rectangle_area_v3(7, 5): $(get_rectangle_area_v3(7, 5))")

# ? anonymous function

function (rectangle_width::Any, rectangle_length::Any)
    return (rectangle_width * rectangle_length)::Any
end

# ? named function expression 🤮

get_rectangle_area_v4::Any = function do_something(rectangle_width::Any, rectangle_length::Any)::Any
    return (rectangle_width * rectangle_length)::Any
end
println("get_rectangle_area_v4(7, 5): $(get_rectangle_area_v4(7, 5))")

# ? function expression with arrow function or named arrow function

get_rectangle_area_v5::Any = (rectangle_width::Any, rectangle_length::Any) -> (begin
    return (rectangle_width * rectangle_length)::Any
end)::Any
println("get_rectangle_area_v5(7, 5): $(get_rectangle_area_v5(7, 5))")

get_rectangle_area_v6::Any = (rectangle_width::Any, rectangle_length::Any) -> (rectangle_width * rectangle_length)::Any
println("get_rectangle_area_v6(7, 5): $(get_rectangle_area_v6(7, 5))")

# ? anonymous arrow function

(rectangle_width::Any, rectangle_length::Any) -> (begin
    return (rectangle_width * rectangle_length)::Any
end)::Any

(rectangle_width::Any, rectangle_length::Any) -> (rectangle_width * rectangle_length)::Any

# ? Passing functions as arguments to other functions

function say_hello(callback_function::Any)::Any
    println("hello")
    callback_function()
end

function say_how_are_you()::Any
    println("how are you ?")
end

say_hello(say_how_are_you)

say_hello(function ()
    println("how are you ?")
end)

say_hello(() -> println("how are you ?"))

# ? Assigning functions to variables or storing them in data structures

get_rectangle_area_v3_copy::Any = function (rectangle_width::Any, rectangle_length::Any)
    return (rectangle_width * rectangle_length)::Any
end
println("get_rectangle_area_v3_copy(7, 5): $(get_rectangle_area_v3_copy(7, 5))")

get_rectangle_area_v5_copy::Any = (rectangle_width::Any, rectangle_length::Any) -> (begin
    return (rectangle_width * rectangle_length)::Any
end)::Any
println("get_rectangle_area_v5_copy(7, 5): $(get_rectangle_area_v5_copy(7, 5))")

get_rectangle_area_v6_copy::Any = (rectangle_width::Any, rectangle_length::Any) -> (rectangle_width * rectangle_length)::Any
println("get_rectangle_area_v6_copy(7, 5): $(get_rectangle_area_v6_copy(7, 5))")


my_array_of_get_rectangle_area_functions::Any = Vector{Any}([
    get_rectangle_area_v1,
    function (rectangle_width::Any, rectangle_length::Any)
        return (rectangle_width * rectangle_length)::Any
    end,
    (rectangle_width::Any, rectangle_length::Any) -> (rectangle_width * rectangle_length)::Any
])
println("my_array_of_get_rectangle_area_functions[1](7, 5): $(my_array_of_get_rectangle_area_functions[1](7, 5))")
println("my_array_of_get_rectangle_area_functions[2](7, 5): $(my_array_of_get_rectangle_area_functions[2](7, 5))")
println("my_array_of_get_rectangle_area_functions[3](7, 5): $(my_array_of_get_rectangle_area_functions[3](7, 5))")
println("try my_array_of_get_rectangle_area_functions[1](7, 5) catch (any_error) nothing end: $(try my_array_of_get_rectangle_area_functions[1](7, 5) catch (any_error) nothing end)")
println("try my_array_of_get_rectangle_area_functions[2](7, 5) catch (any_error) nothing end: $(try my_array_of_get_rectangle_area_functions[2](7, 5) catch (any_error) nothing end)")
println("try my_array_of_get_rectangle_area_functions[3](7, 5) catch (any_error) nothing end: $(try my_array_of_get_rectangle_area_functions[3](7, 5) catch (any_error) nothing end)")

function exponentiation(a::Any, b::Any)::Any
    return (a ^ b)::Any
end

simple_calculator::Any = Dict{String, Any}(
    "exponentiation" => exponentiation,
    "addition" => function (a::Any, b::Any)
        return (a + b)::Any
    end,
    "subtraction" => (a::Any, b::Any) -> (a - b)::Any
)
println("simple_calculator[\"exponentiation\"](2, 4): $(simple_calculator["exponentiation"](2, 4))")
println("simple_calculator[\"addition\"](9, 3): $(simple_calculator["addition"](9, 3))")
println("simple_calculator[\"subtraction\"](35, 8): $(simple_calculator["subtraction"](35, 8))")
println("try simple_calculator[\"exponentiation\"](2, 4) catch (any_error) nothing end: $(try simple_calculator["exponentiation"](2, 4) catch (any_error) nothing end)")
println("try simple_calculator[\"addition\"](9, 3) catch (any_error) nothing end: $(try simple_calculator["addition"](9, 3) catch (any_error) nothing end)")
println("try simple_calculator[\"subtraction\"](35, 8) catch (any_error) nothing end: $(try simple_calculator["subtraction"](35, 8) catch (any_error) nothing end)")

# ? Returning functions as values from other functions

function multiply_v1(a::Any)::Any
    function multiply_by(b::Any)::Any
        return (a * b)::Any
    end
    return (multiply_by)::Any
end
multiply_by2::Any = multiply_v1(2)
multiply_by2_result::Any = multiply_by2(10)
println("multiply_by2(10): $(multiply_by2_result)")

function multiply_v2(a::Any)::Any
    local multiply_by::Any = function (b::Any)
        return (a * b)::Any
    end
    return (multiply_by)::Any
end
multiply_by3::Any = multiply_v2(3)
multiply_by3_result::Any = multiply_by3(10)
println("multiply_by3(10): $(multiply_by3_result)")

function multiply_v3(a::Any)::Any
    local multiply_by::Any = (b::Any) -> (begin
        return (a * b)::Any
    end)::Any
    return (multiply_by)::Any
end
multiply_by4::Any = multiply_v3(4)
multiply_by4_result::Any = multiply_by4(10)
println("multiply_by4(10): $(multiply_by4_result)")

function multiply_v4(a::Any)::Any
    local multiply_by::Any = (b::Any) -> (a * b)::Any
    return (multiply_by)::Any
end
multiply_by5::Any = multiply_v4(5)
multiply_by5_result::Any = multiply_by5(10)
println("multiply_by5(10): $(multiply_by5_result)")

multiply_v5::Any = function (a::Any)
    local multiply_by::Any = function (b::Any)
        return (a * b)::Any
    end
    return (multiply_by)::Any
end
multiply_by6::Any = multiply_v5(6)
multiply_by6_result::Any = multiply_by6(10)
println("multiply_by6(10): $(multiply_by6_result)")

multiply_v6::Any = function (a::Any)
    local multiply_by::Any = (b::Any) -> (begin
        return (a * b)::Any
    end)::Any
    return (multiply_by)::Any
end
multiply_by7::Any = multiply_v6(7)
multiply_by7_result::Any = multiply_by7(10)
println("multiply_by7(10): $(multiply_by7_result)")

multiply_v7::Any = function (a::Any)
    local multiply_by::Any = (b::Any) -> (a * b)::Any
    return (multiply_by)::Any
end
multiply_by8::Any = multiply_v7(8)
multiply_by8_result::Any = multiply_by8(10)
println("multiply_by8(10): $(multiply_by8_result)")

multiply_v8::Any = (a::Any) -> (begin
    local multiply_by::Any = function (b::Any)
        return (a * b)::Any
    end
    return (multiply_by)::Any
end)::Any
multiply_by9::Any = multiply_v8(9)
multiply_by9_result::Any = multiply_by9(10)
println("multiply_by9(10): $(multiply_by9_result)")

multiply_v9::Any = (a::Any) -> (begin
    local multiply_by::Any = (b::Any) -> (begin
        return (a * b)::Any
    end)::Any
    return (multiply_by)::Any
end)::Any
multiply_by10::Any = multiply_v9(10)
multiply_by10_result::Any = multiply_by10(10)
println("multiply_by10(10): $(multiply_by10_result)")

multiply_v10::Any = (a::Any) -> (begin
    local multiply_by::Any = (b::Any) -> (a * b)::Any
    return (multiply_by)::Any
end)::Any
multiply_by11::Any = multiply_v10(11)
multiply_by11_result::Any = multiply_by11(10)
println("multiply_by11(10): $(multiply_by11_result)")

multiply_v11::Any = function (a::Any)
    return (function (b::Any)
        return (a * b)::Any
    end)::Any
end
multiply_by12::Any = multiply_v11(12)
multiply_by12_result::Any = multiply_by12(10)
println("multiply_by12(10): $(multiply_by12_result)")

multiply_v12::Any = function (a::Any)
    return ((b::Any) -> (begin
        return (a * b)::Any
    end)::Any)::Any
end
multiply_by13::Any = multiply_v12(13)
multiply_by13_result::Any = multiply_by13(10)
println("multiply_by13(10): $(multiply_by13_result)")

multiply_v13::Any = function (a::Any)
    return ((b::Any) -> (a * b)::Any)::Any
end
multiply_by14::Any = multiply_v13(14)
multiply_by14_result::Any = multiply_by14(10)
println("multiply_by14(10): $(multiply_by14_result)")

multiply_v14::Any = (a::Any) -> (begin
    return (function (b::Any)
        return (a * b)::Any
    end)::Any
end)::Any
multiply_by15::Any = multiply_v14(15)
multiply_by15_result::Any = multiply_by15(10)
println("multiply_by15(10): $(multiply_by15_result)")

multiply_v15::Any = (a::Any) -> (begin
    return ((b::Any) -> (begin
        return (a * b)::Any
    end)::Any)::Any
end)::Any
multiply_by16::Any = multiply_v15(16)
multiply_by16_result::Any = multiply_by16(10)
println("multiply_by16(10): $(multiply_by16_result)")

multiply_v16::Any = (a::Any) -> (begin
    return ((b::Any) -> (a * b)::Any)::Any
end)::Any
multiply_by17::Any = multiply_v16(17)
multiply_by17_result::Any = multiply_by17(10)
println("multiply_by17(10): $(multiply_by17_result)")

multiply_v17::Any = (a::Any) -> ((b::Any) -> (a * b)::Any)::Any
multiply_by18::Any = multiply_v17(18)
multiply_by18_result::Any = multiply_by18(10)
println("multiply_by18(10): $(multiply_by18_result)")

File: codes/javascript/src/007_functions.js
/* eslint-disable dot-notation */
/* eslint-disable no-console */
/* eslint-disable object-shorthand */
/* eslint-disable arrow-body-style */
/* eslint-disable prefer-arrow-callback */
/* eslint-disable func-names */

// ? function statement or function declaration

function getRectangleAreaV1(rectangleWidth, rectangleLength) {
    return (rectangleWidth * rectangleLength);
}
console.log(`getRectangleAreaV1(7, 5): ${getRectangleAreaV1(7, 5)}`);

// ? function expression

const getRectangleAreaV2 = function (rectangleWidth, rectangleLength) {
    return (rectangleWidth * rectangleLength);
};
console.log(`getRectangleAreaV2(7, 5): ${getRectangleAreaV2(7, 5)}`);

// ? function expression with arrow function or named arrow function

const getRectangleAreaV3 = (rectangleWidth, rectangleLength) => {
    return (rectangleWidth * rectangleLength);
};
console.log(`getRectangleAreaV3(7, 5): ${getRectangleAreaV3(7, 5)}`);

const getRectangleAreaV4 = (rectangleWidth, rectangleLength) => (rectangleWidth * rectangleLength);
console.log(`getRectangleAreaV4(7, 5): ${getRectangleAreaV4(7, 5)}`);

// ? named function expression 🤮

const getRectangleAreaV5 = function doSomething(rectangleWidth, rectangleLength) {
    return (rectangleWidth * rectangleLength);
};
console.log(`getRectangleAreaV5(7, 5): ${getRectangleAreaV5(7, 5)}`);

// ? anonymous function

// function (rectangleWidth, rectangleLength) {
//     return (rectangleWidth * rectangleLength);
// }

// ? anonymous arrow function

// (rectangleWidth, rectangleLength) => {
//     return (rectangleWidth * rectangleLength);
// };
// (rectangleWidth, rectangleLength) => (rectangleWidth * rectangleLength)

/*
    ? In JavaScript, a function is a first-class citizen, which means it supports:
    ? • Passing functions as arguments to other functions
    ? • Assigning functions to variables or storing them in data structures
    ? • Returning functions as values from other functions
*/

// ? Passing functions as arguments to other functions

function sayHello(callbackFunction) {
    console.log("hello");
    callbackFunction();
}

function sayHowAreYou() {
    console.log("how are you?");
}

sayHello(sayHowAreYou);

sayHello(function () {
    console.log("how are you?");
});

sayHello(() => {
    console.log("how are you?");
});

sayHello(() => console.log("how are you?"));

// ? Assigning functions to variables or storing them in data structures

const getRectangleAreaV2Copy = function (rectangleWidth, rectangleLength) {
    return (rectangleWidth * rectangleLength);
};
console.log(`getRectangleAreaV2Copy(7, 5): ${getRectangleAreaV2Copy(7, 5)}`);

const getRectangleAreaV3Copy = (rectangleWidth, rectangleLength) => {
    return (rectangleWidth * rectangleLength);
};
console.log(`getRectangleAreaV3Copy(7, 5): ${getRectangleAreaV3Copy(7, 5)}`);

const getRectangleAreaV4Copy = (rectangleWidth, rectangleLength) => (rectangleWidth * rectangleLength);
console.log(`getRectangleAreaV4Copy(7, 5): ${getRectangleAreaV4Copy(7, 5)}`);

const myArrayOfGetRectangleAreaFunctions = [
    getRectangleAreaV1,
    function (rectangleWidth, rectangleLength) {
        return (rectangleWidth * rectangleLength);
    },
    (rectangleWidth, rectangleLength) => {
        return (rectangleWidth * rectangleLength);
    },
    (rectangleWidth, rectangleLength) => (rectangleWidth * rectangleLength)
];
console.log(`myArrayOfGetRectangleAreaFunctions[0](7, 5): ${myArrayOfGetRectangleAreaFunctions[0](7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions[1](7, 5): ${myArrayOfGetRectangleAreaFunctions[1](7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions[2](7, 5): ${myArrayOfGetRectangleAreaFunctions[2](7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions[3](7, 5): ${myArrayOfGetRectangleAreaFunctions[3](7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions?.[0]?.(7, 5): ${myArrayOfGetRectangleAreaFunctions?.[0]?.(7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions?.[1]?.(7, 5): ${myArrayOfGetRectangleAreaFunctions?.[1]?.(7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions?.[2]?.(7, 5): ${myArrayOfGetRectangleAreaFunctions?.[2]?.(7, 5)}`);
console.log(`myArrayOfGetRectangleAreaFunctions?.[3]?.(7, 5): ${myArrayOfGetRectangleAreaFunctions?.[3]?.(7, 5)}`);

function exponentiation(a, b) {
    return (a ** b);
}

const simpleCalculator = {
    exponentiation,
    multiplication: function (a, b) {
        return (a * b);
    },
    division: (a, b) => {
        return (a / b);
    },
    addition: (a, b) => (a + b),
    subtraction(a, b) {
        return (a - b);
    },
};
console.log(`simpleCalculator["exponentiation"](2, 4): ${simpleCalculator["exponentiation"](2, 4)}`);
console.log(`simpleCalculator["multiplication"](7, 5): ${simpleCalculator["multiplication"](7, 5)}`);
console.log(`simpleCalculator["division"](81, 9): ${simpleCalculator["division"](81, 9)}`);
console.log(`simpleCalculator["addition"](9, 3): ${simpleCalculator["addition"](9, 3)}`);
console.log(`simpleCalculator["subtraction"](35, 8): ${simpleCalculator["subtraction"](35, 8)}`);
console.log(`simpleCalculator.exponentiation(2, 4): ${simpleCalculator.exponentiation(2, 4)}`);
console.log(`simpleCalculator.multiplication(7, 5): ${simpleCalculator.multiplication(7, 5)}`);
console.log(`simpleCalculator.division(81, 9): ${simpleCalculator.division(81, 9)}`);
console.log(`simpleCalculator.addition(9, 3): ${simpleCalculator.addition(9, 3)}`);
console.log(`simpleCalculator.subtraction(35, 8): ${simpleCalculator.subtraction(35, 8)}`);
console.log(`simpleCalculator?.["exponentiation"]?.(2, 4): ${simpleCalculator?.["exponentiation"]?.(2, 4)}`);
console.log(`simpleCalculator?.["multiplication"]?.(7, 5): ${simpleCalculator?.["multiplication"]?.(7, 5)}`);
console.log(`simpleCalculator?.["division"]?.(81, 9): ${simpleCalculator?.["division"]?.(81, 9)}`);
console.log(`simpleCalculator?.["addition"]?.(9, 3): ${simpleCalculator?.["addition"]?.(9, 3)}`);
console.log(`simpleCalculator?.["subtraction"]?.(35, 8): ${simpleCalculator?.["subtraction"]?.(35, 8)}`);
console.log(`simpleCalculator?.exponentiation?.(2, 4): ${simpleCalculator?.exponentiation?.(2, 4)}`);
console.log(`simpleCalculator?.multiplication?.(7, 5): ${simpleCalculator?.multiplication?.(7, 5)}`);
console.log(`simpleCalculator?.division?.(81, 9): ${simpleCalculator?.division?.(81, 9)}`);
console.log(`simpleCalculator?.addition?.(9, 3): ${simpleCalculator?.addition?.(9, 3)}`);
console.log(`simpleCalculator?.subtraction?.(35, 8): ${simpleCalculator?.subtraction?.(35, 8)}`);

// ? Returning functions as values from other functions

function multiplyV1(a) {
    function multiplyBy(b) {
        return (a * b);
    }
    return multiplyBy;
}
const multiplyBy2 = multiplyV1(2);
const multiplyBy2Result = multiplyBy2(10);
console.log(`multiplyBy2(10): ${multiplyBy2Result}`);

function multiplyV2(a) {
    const multiplyBy = function (b) {
        return (a * b);
    };
    return multiplyBy;
}
const multiplyBy3 = multiplyV2(3);
const multiplyBy3Result = multiplyBy3(10);
console.log(`multiplyBy3(10): ${multiplyBy3Result}`);

function multiplyV3(a) {
    const multiplyBy = (b) => {
        return (a * b);
    };
    return multiplyBy;
}
const multiplyBy4 = multiplyV3(4);
const multiplyBy4Result = multiplyBy4(10);
console.log(`multiplyBy4(10): ${multiplyBy4Result}`);

function multiplyV4(a) {
    const multiplyBy = (b) => (a * b);
    return multiplyBy;
}
const multiplyBy5 = multiplyV4(5);
const multiplyBy5Result = multiplyBy5(10);
console.log(`multiplyBy5(10): ${multiplyBy5Result}`);

function multiplyV5(a) {
    return (b) => (a * b);
}
const multiplyBy6 = multiplyV5(6);
const multiplyBy6Result = multiplyBy6(10);
console.log(`multiplyBy6(10): ${multiplyBy6Result}`);

function multiplyV6(a) {
    return function (b) {
        return (a * b);
    };
}
const multiplyBy7 = multiplyV6(7);
const multiplyBy7Result = multiplyBy7(10);
console.log(`multiplyBy7(10): ${multiplyBy7Result}`);

function multiplyV7(a) {
    return (b) => {
        return (a * b);
    };
}
const multiplyBy8 = multiplyV7(8);
const multiplyBy8Result = multiplyBy8(10);
console.log(`multiplyBy8(10): ${multiplyBy8Result}`);

const multiplyV8 = function (a) {
    function multiplyBy(b) {
        return (a * b);
    }
    return multiplyBy;
};
const multiplyBy9 = multiplyV8(9);
const multiplyBy9Result = multiplyBy9(10);
console.log(`multiplyBy9(10): ${multiplyBy9Result}`);

const multiplyV9 = function (a) {
    const multiplyBy = function (b) {
        return (a * b);
    };
    return multiplyBy;
};
const multiplyBy10 = multiplyV9(10);
const multiplyBy10Result = multiplyBy10(10);
console.log(`multiplyBy10(10): ${multiplyBy10Result}`);

const multiplyV10 = function (a) {
    const multiplyBy = (b) => {
        return (a * b);
    };
    return multiplyBy;
};
const multiplyBy11 = multiplyV10(11);
const multiplyBy11Result = multiplyBy11(10);
console.log(`multiplyBy11(10): ${multiplyBy11Result}`);

const multiplyV11 = function (a) {
    const multiplyBy = (b) => (a * b);
    return multiplyBy;
};
const multiplyBy12 = multiplyV11(12);
const multiplyBy12Result = multiplyBy12(10);
console.log(`multiplyBy12(10): ${multiplyBy12Result}`);

const multiplyV12 = function (a) {
    return (b) => (a * b);
};
const multiplyBy13 = multiplyV12(13);
const multiplyBy13Result = multiplyBy13(10);
console.log(`multiplyBy13(10): ${multiplyBy13Result}`);

const multiplyV13 = function (a) {
    return function (b) {
        return (a * b);
    };
};
const multiplyBy14 = multiplyV13(14);
const multiplyBy14Result = multiplyBy14(10);
console.log(`multiplyBy14(10): ${multiplyBy14Result}`);

const multiplyV14 = function (a) {
    return (b) => {
        return (a * b);
    };
};
const multiplyBy15 = multiplyV14(15);
const multiplyBy15Result = multiplyBy15(10);
console.log(`multiplyBy15(10): ${multiplyBy15Result}`);

const multiplyV15 = (a) => {
    function multiplyBy(b) {
        return (a * b);
    }
    return multiplyBy;
};
const multiplyBy16 = multiplyV15(16);
const multiplyBy16Result = multiplyBy16(10);
console.log(`multiplyBy16(10): ${multiplyBy16Result}`);

const multiplyV16 = (a) => {
    const multiplyBy = function (b) {
        return (a * b);
    };
    return multiplyBy;
};
const multiplyBy17 = multiplyV16(17);
const multiplyBy17Result = multiplyBy17(10);
console.log(`multiplyBy17(10): ${multiplyBy17Result}`);

const multiplyV17 = (a) => {
    const multiplyBy = (b) => {
        return (a * b);
    };
    return multiplyBy;
};
const multiplyBy18 = multiplyV17(18);
const multiplyBy18Result = multiplyBy18(10);
console.log(`multiplyBy18(10): ${multiplyBy18Result}`);

const multiplyV18 = (a) => {
    const multiplyBy = (b) => (a * b);
    return multiplyBy;
};
const multiplyBy19 = multiplyV18(19);
const multiplyBy19Result = multiplyBy19(10);
console.log(`multiplyBy19(10): ${multiplyBy19Result}`);

const multiplyV19 = (a) => {
    return (b) => (a * b);
};
const multiplyBy20 = multiplyV19(20);
const multiplyBy20Result = multiplyBy20(10);
console.log(`multiplyBy20(10): ${multiplyBy20Result}`);

const multiplyV20 = (a) => {
    return function (b) {
        return (a * b);
    };
};
const multiplyBy21 = multiplyV20(21);
const multiplyBy21Result = multiplyBy21(10);
console.log(`multiplyBy21(10): ${multiplyBy21Result}`);

const multiplyV21 = (a) => {
    return (b) => {
        return (a * b);
    };
};
const multiplyBy22 = multiplyV21(22);
const multiplyBy22Result = multiplyBy22(10);
console.log(`multiplyBy22(10): ${multiplyBy22Result}`);

const multiplyV22 = (a) => (b) => (a * b);
const multiplyBy23 = multiplyV22(23);
const multiplyBy23Result = multiplyBy23(10);
console.log(`multiplyBy23(10): ${multiplyBy23Result}`);
